### PR TITLE
feat: [JetPack Compose] Refactor compose handlers

### PR DESCRIPTION
**Pull request by sravanmedarapu**
- Moved FindElement, FindElements and GetAttribute handlers to corresponding classes
- Removed redundant ComposeDriver, EspressoDriver, AppDriver classes.

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetAttribute.kt [hunk 1]
```diff
@@ -32,11 +39,70 @@ class GetAttribute : RequestHandler<AppiumParams, String?> {
 
         // Map attributeName to ENUM attribute
         ViewAttributesEnum.values().find { attributeName.equals(it.toString(), ignoreCase = true) }
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetAttribute.kt:41]: the list of available attributes for espresso and compose is different, so I don't think we should unify that

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ViewAttributesEnum.kt [hunk 2]
```diff
@@ -17,35 +17,97 @@
 package io.appium.espressoserver.lib.model
 
 enum class ViewAttributesEnum {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ViewAttributesEnum.kt:19]: Shall we rename it to just AttributesEnum?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ViewAttributesEnum.kt [hunk 3]
```diff
@@ -17,35 +17,97 @@
...
+    NO_OVERLAPS,
+    PACKAGE,
+    VISIBLE;
 
     override fun toString(): String {
         return this.name.replace("_", "-").lowercase()
     }
 }
+
+class EspressoAttributes(override var attributes: List<ViewAttributesEnum> =
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/ViewAttributesEnum.kt:55]: 👍  Nicely done

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/viewmatcher/WithXPath.kt [hunk 4]
```diff
@@ -17,17 +17,17 @@ package io.appium.espressoserver.lib.viewmatcher
...
 import org.hamcrest.TypeSafeMatcher
 
-fun fetchIncludedAttributes(xpath: String): Set<ViewAttributesEnum>? {
+fun fetchIncludedAttributes(xpath: String): Set<AttributesEnum>? {
     if (xpath.contains("@*")) {
         return null
     }
 
-    return ViewAttributesEnum.values().fold(mutableSetOf()) { acc, value ->
+    return AttributesEnum.values().fold(mutableSetOf()) { acc, value ->
```
[the remark this answers is not in the document]
Comment (sravanmedarapu) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/viewmatcher/WithXPath.kt:30]: hmm, I am not familiar with this functionality :)
By the looks, seems to a generic logic triggered as part of findElement.
In favour of limiting the scope of the PR, would prefer to leave as is for now if it's not major and address separately if needed.

[the remark this answers is not in the document]
Comment (sravanmedarapu) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/viewmatcher/WithXPath.kt:30]: I agree on changing to `EspressoAttributes().attributes` wouldn't make any difference as `AttributesEnum.values()` and `EspressoAttributes().attributes` returns same at the moment.
  
My concern is more on, why we are referring to EspressoDriver specific attributes, in generic logic which seems gets triggered for compose and espresso. Please correct me if I didn't get it right.

[the remark this answers is not in the document]
Comment (sravanmedarapu) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/viewmatcher/WithXPath.kt:30]: Ha ha agree, I was just not confident :)

**Comment by mykola-mokhnach** [comment]
Could you also check tests? It looks like some of them fail now:

```
2021-10-21T19:35:48.7687330Z dbug WD Proxy Proxying [GET /wd/hub/session/2834b5cc-bc25-4c68-b7f6-53deb5433486/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc] to [GET http://127.0.0.1:8300/session/89c67a2a-923d-49cd-8fee-c2fd3ab354bb/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc] with no body
2021-10-21T19:35:48.9454520Z info WD Proxy Got response with status 404: {"id":"6500ee58-e212-4518-9987-dd237779e084","sessionId":"89c67a2a-923d-49cd-8fee-c2fd3ab354bb","value":{"error":"unknown method","message":"Espresso supports only 'bounds', 'class', 'clickable', 'content-desc', 'enabled', 'focused', 'index', 'password', 'resource-id', 'scrollable', 'selected', 'text', 'view-tag', 'adapters', 'adapter-type', 'checkable', 'checked', 'focusable', 'hint', 'instance', 'long-clickable', 'no-ellipsized-text', 'no-multiline-buttons', 'no-overlaps', 'package', 'visible' attributes but 'content-desc' is given instead","stacktrace":"io.appium.espressoserver.lib.handlers.exceptions.NotYetImplementedException: Espresso supports only 'bounds', 'class', 'clickable', 'content-desc', 'enabled', 'focused', 'index', 'password', 'resource-id', 'scrollable', 'selected', 'text', 'view-tag', 'adapters', 'adapter-type', 'checkable', 'checked', 'focusable', 'hint', 'instance', 'long-clickable', 'no-ellipsized-text', 'no-multiline-buttons', 'no-overlaps', 'package', 'visible' attributes but 'conte...
2021-10-21T19:35:48.9624800Z dbug W3C Matched W3C error code 'unknown method' to UnknownMethodError
2021-10-21T19:35:48.9637620Z info HTTP <-- GET /wd/hub/session/2834b5cc-bc25-4c68-b7f6-53deb5433486/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc 500 197 ms - 494
2021-10-21T19:35:48.9639100Z info HTTP 
2021-10-21T19:35:48.9753980Z       1) should get the 'content-desc' of a View
```

**Comment by sravanmedarapu** [comment]
> Could you also check tests? It looks like some of them fail now:
> 
> ```
> 2021-10-21T19:35:48.7687330Z dbug WD Proxy Proxying [GET /wd/hub/session/2834b5cc-bc25-4c68-b7f6-53deb5433486/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc] to [GET http://127.0.0.1:8300/session/89c67a2a-923d-49cd-8fee-c2fd3ab354bb/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc] with no body
> 2021-10-21T19:35:48.9454520Z info WD Proxy Got response with status 404: {"id":"6500ee58-e212-4518-9987-dd237779e084","sessionId":"89c67a2a-923d-49cd-8fee-c2fd3ab354bb","value":{"error":"unknown method","message":"Espresso supports only 'bounds', 'class', 'clickable', 'content-desc', 'enabled', 'focused', 'index', 'password', 'resource-id', 'scrollable', 'selected', 'text', 'view-tag', 'adapters', 'adapter-type', 'checkable', 'checked', 'focusable', 'hint', 'instance', 'long-clickable', 'no-ellipsized-text', 'no-multiline-buttons', 'no-overlaps', 'package', 'visible' attributes but 'content-desc' is given instead","stacktrace":"io.appium.espressoserver.lib.handlers.exceptions.NotYetImplementedException: Espresso supports only 'bounds', 'class', 'clickable', 'content-desc', 'enabled', 'focused', 'index', 'password', 'resource-id', 'scrollable', 'selected', 'text', 'view-tag', 'adapters', 'adapter-type', 'checkable', 'checked', 'focusable', 'hint', 'instance', 'long-clickable', 'no-ellipsized-text', 'no-multiline-buttons', 'no-overlaps', 'package', 'visible' attributes but 'conte...
> 2021-10-21T19:35:48.9624800Z dbug W3C Matched W3C error code 'unknown method' to UnknownMethodError
> 2021-10-21T19:35:48.9637620Z info HTTP <-- GET /wd/hub/session/2834b5cc-bc25-4c68-b7f6-53deb5433486/element/775beaec-84e6-4329-bda0-195f6b2c1381/attribute/content-desc 500 197 ms - 494
> 2021-10-21T19:35:48.9639100Z info HTTP 
> 2021-10-21T19:35:48.9753980Z       1) should get the 'content-desc' of a View
> ```

Thanks for the headsup, addressed.

**Pull request merged**